packages/integration-tests/src/test-e2e/create_amplify.test.ts

```diff
@@ -7,19 +7,87 @@ import { after, afterEach, before, beforeEach, describe, it } from 'node:test';
 import assert from 'assert';
 import { glob } from 'glob';
 
-void describe('create-amplify script', () => {
-  before(async () => {
-    // start a local npm proxy and publish the current codebase to the proxy
-    await execa('npm', ['run', 'clean:npm-proxy'], { stdio: 'inherit' });
-    await execa('npm', ['run', 'vend'], { stdio: 'inherit' });
+type PackageManagerExecutable = 'npm' | 'yarn' | 'yarn-stable' | 'pnpm';
 
-    // nuke the npx cache to ensure we are installing packages from the npm proxy
+const packageManagerSetup = async (
```

Comment:
at some point this will need to be shared between test files, so worth extracting `class PackageManagerConfigurator` or something like this in separate file.

In general I'd suggest to start packing all these package manager customization behind some abstractions. They don't have to be perfect at this stage but should drive component breakdown, so later changes to get this to main will be easier.